server/lib/util.js

```diff
@@ -26,16 +27,31 @@ module.exports.take = take;
  * @todo - allow prexisting conditions to be passed in
  */
 
-function queryCondition(sql, params, excludeWhere) {
+function queryCondition(sql, params, excludeWhere, dateConditon) {
   let conditions = [];
 
+  let dateParams = {
+    dateFrom: new Date(params.dateFrom),
+    dateTo: new Date(params.dateTo)
+  };
+
+  if (Object.keys(params).length === 0) {
+    excludeWhere = true;
+  }
+
+  // remove date in params
+  delete params.dateFrom;
```

Comment:
This is incredibly specific to your module's code, but you've put it in a public library.  This is really bad practice.  Could you remove these parameters before calling this function somehow?